OCPBUGS#30031: Doc improvements related to "Monitoring" section

sr1kar99 opened the pull request:
Version(s):
4.12+

Issue:
[OCPBUGS-30031](https://issues.redhat.com/browse/OCPBUGS-30031)


Link to docs preview:
[Preview](https://72264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms#lvms-monitoring-using-lvms_logical-volume-manager-storage)


QE review:
- [x] QE has approved this change.

sr1kar99 commented:
@jakobmoellerdev , @suleymanakbas91 
Can you please review this PR?
Thanks.

suleymanakbas91 commented:
/lgtm

sr1kar99 commented:
/label peer-review-needed

dfitzmau commented:
/remove-label peer-review-needed

/label peer-review-in-progress

dfitzmau commented on modules/lvms-download-log-files-and-diagnostics.adoc:
```diff
@@ -8,7 +8,7 @@
 When {lvms} is unable to automatically resolve a problem, use the must-gather tool to collect the log files and diagnostic information so that you or the Red Hat Support can review the problem and determine a solution.
 
-* Run the must-gather command from the client connected to {lvms} cluster by running the following command:
+* Run the must-gather command from the client connected to {lvms} cluster:
```
```suggestion
* Run the `must-gather` command from the client connected to {lvms} cluster:
```

[Example](https://docs.openshift.com/container-platform/4.15/nodes/cma/nodes-cma-autoscaling-custom-debugging.html) of highlighting in other docs.

sr1kar99 replied:
Updated as per the suggestion.

dfitzmau commented on modules/lvms-monitoring-logical-volume-manager-operator.adoc:
```diff
@@ -6,42 +6,56 @@
+To enable cluster monitoring, you must add the following label in the namespace where you have installed {lvms}:
+[source,text]
+----
+openshift.io/cluster-monitoring=true
```
I'm not fully sure on the context of where to apply this command. Add to a config file, assign to an environment variable, or launch a tool and specify the command there?

Examples from the Adding custom metrics link:

![Screenshot from 2024-03-08 10-24-48](https://github.com/openshift/openshift-docs/assets/57954076/0fa2dae9-8fd6-4062-98a4-b239e8d59752)
![Screenshot from 2024-03-08 10-24-51](https://github.com/openshift/openshift-docs/assets/57954076/c8a813fb-6153-4226-b220-d46541e8e9c9)

sr1kar99 replied:
This label must be added in a namespace:

![Screenshot from 2024-03-08 16-43-01](https://github.com/openshift/openshift-docs/assets/148952239/1cbcfa96-3a3c-4330-a560-463966321630)

Click *Edit* to add a new label.

dfitzmau replied:
Ah, OK. Thanks. This makes sense.

dfitzmau commented on modules/lvms-monitoring-logical-volume-manager-operator.adoc:
```diff
@@ -6,42 +6,56 @@
+The following table describes the `topolvm` metrics:
+
+.`topolvm` metrics
+[%autowidth,frame="topbot",options="header"]
```
I like this table. I know "topbot" is valid a HTML component but [Asciidoctor](https://docs.asciidoctor.org/asciidoc/latest/tables/borders/) does not list it. Or platforms can be a bit funny when it comes to supporting cetain syntax. I do not want to kill your formatting buzz but asking a merge reviewer to check how platform docs render on Pantheon would be worthwile.

dfitzmau commented on modules/lvms-monitoring-logical-volume-manager-operator.adoc:
```diff
@@ -6,42 +6,56 @@
-{lvms} sends the following alerts about the usage of the thin pool and volume group when utilization crosses a certain value:
+When the thin pool and volume group reach maximum storage capacity, further operations fail. This can lead to data loss.
+
+{lvms} sends the following alerts when the utilization of the thin pool and volume group crosses a certain value:
```
![Screenshot from 2024-03-08 10-32-21](https://github.com/openshift/openshift-docs/assets/57954076/a2dcbee9-ba09-4e51-ab26-90eff2f7ac44)

sr1kar99 replied:
Updated as per the suggestion.

dfitzmau commented on modules/lvms-monitoring-logical-volume-manager-operator.adoc:
```diff
@@ -6,42 +6,56 @@
-{lvms} sends the following alerts about the usage of the thin pool and volume group when utilization crosses a certain value:
+When the thin pool and volume group reach maximum storage capacity, further operations fail. This can lead to data loss.
+
+{lvms} sends the following alerts when the utilization of the thin pool and volume group crosses a certain value:
```
```suggestion
{lvms} sends the following alerts when the usage of the thin pool and volume group exceeds a certain value:
```

sr1kar99 replied:
Updated as per the suggestion.

dfitzmau commented on modules/lvms-monitoring-logical-volume-manager-operator.adoc:
```diff
@@ -6,42 +6,56 @@
+|`VolumeGroupUsageAtThresholdCritical`|This alert is triggered when both the volume group and thin pool utilization cross 85% on nodes. In this case, the volume group is critically full. Data deletion or volume group expansion is required.
 |`ThinPoolDataUsageAtThresholdNearFull`|This alert is triggered when the thin pool data utilization in the volume group crosses 75% on nodes. Data deletion or thin pool expansion is required.
-|`ThinPoolDataUsageAtThresholdCritical`|This alert is triggered when  the thin pool data utilization in the volume group crosses 85% on nodes. Data deletion or thin pool expansion is required.
+|`ThinPoolDataUsageAtThresholdCritical`|This alert is triggered when the thin pool data utilization in the volume group crosses 85% on nodes. Data deletion or thin pool expansion is required.
```
```suggestion
|`ThinPoolDataUsageAtThresholdCritical`|This alert is triggered when the thin pool data usage in the volume group exceeds 85% on nodes. Data deletion or thin pool expansion is required.
```

sr1kar99 replied:
Updated as per the suggestion.

sr1kar99 commented:
> Hi @sr1kar99 . Excellent work to you and the engineers. Discussing the limitations when limtis are reached is a tough artform. I added comments inline. I like your "topbot" styling use but I'm not sure how our platform infrastructure will handle it come publication.
> 
> Let me know if you would like to present the styling to the wider team and get their viewpoint.
> 
> /remove-label peer-review-in-progress
> 
> /label peer-review-done

Hi @dfitzmau 
Thank you for the review. 
Regarding the topbot frame, I was trying to explore the styling options, but not sure if we can use it as I though it could lead to inconsistent styling of tables. 
Do we have guidelines on this?

dfitzmau commented:
> > Hi @sr1kar99 . Excellent work to you and the engineers. Discussing the limitations when limits are reached is a tough artform. I added comments inline. I like your "topbot" styling use but I'm not sure how our platform infrastructure will handle it come publication.
> > Let me know if you would like to present the styling to the wider team and get their viewpoint.
> > /remove-label peer-review-in-progress
> > /label peer-review-done
> 
> Hi @dfitzmau Thank you for the review. Regarding the topbot frame, I was trying to explore the styling options, but not sure if we can use it as I thought it could lead to inconsistent styling of tables. Do we have guidelines on this?

Hi @sr1kar99 . Tbale gudiance is lacking in our [guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#formatting) doc. Usually when we something listed in the Asciidoc documentation, we can experiment. I'll add something to the team docs channel.

dfitzmau commented:
> > > Hi @sr1kar99 . Excellent work to you and the engineers. Discussing the limitations when limits are reached is a tough artform. I added comments inline. I like your "topbot" styling use but I'm not sure how our platform infrastructure will handle it come publication.
> > > Let me know if you would like to present the styling to the wider team and get their viewpoint.
> > > /remove-label peer-review-in-progress
> > > /label peer-review-done
> > 
> > 
> > Hi @dfitzmau Thank you for the review. Regarding the topbot frame, I was trying to explore the styling options, but not sure if we can use it as I thought it could lead to inconsistent styling of tables. Do we have guidelines on this?
> 
> Hi @sr1kar99 . Tbale gudiance is lacking in our [guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#formatting) doc. Usually when we something listed in the Asciidoc documentation, we can experiment. I'll add something to the team docs channel.

I've asked [here](https://redhat-internal.slack.com/archives/C02KBD8A4UF/p1709897478675109).

sr1kar99 commented:
> > > > Hi @sr1kar99 . Excellent work to you and the engineers. Discussing the limitations when limits are reached is a tough artform. I added comments inline. I like your "topbot" styling use but I'm not sure how our platform infrastructure will handle it come publication.
> > > > Let me know if you would like to present the styling to the wider team and get their viewpoint.
> > > > /remove-label peer-review-in-progress
> > > > /label peer-review-done
> > > 
> > > 
> > > Hi @dfitzmau Thank you for the review. Regarding the topbot frame, I was trying to explore the styling options, but not sure if we can use it as I thought it could lead to inconsistent styling of tables. Do we have guidelines on this?
> > 
> > 
> > Hi @sr1kar99 . Tbale gudiance is lacking in our [guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#formatting) doc. Usually when we something listed in the Asciidoc documentation, we can experiment. I'll add something to the team docs channel.
> 
> I've asked [here](https://redhat-internal.slack.com/archives/C02KBD8A4UF/p1709897478675109).

Hi @dfitzmau 
Thanks very much for adding this query to the docs channel. 
As discussed, I have removed the frame option.

dfitzmau commented:
> > > > > Hi @sr1kar99 . Excellent work to you and the engineers. Discussing the limitations when limits are reached is a tough artform. I added comments inline. I like your "topbot" styling use but I'm not sure how our platform infrastructure will handle it come publication.
> > > > > Let me know if you would like to present the styling to the wider team and get their viewpoint.
> > > > > /remove-label peer-review-in-progress
> > > > > /label peer-review-done
> > > > 
> > > > 
> > > > Hi @dfitzmau Thank you for the review. Regarding the topbot frame, I was trying to explore the styling options, but not sure if we can use it as I thought it could lead to inconsistent styling of tables. Do we have guidelines on this?
> > > 
> > > 
> > > Hi @sr1kar99 . Tbale gudiance is lacking in our [guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#formatting) doc. Usually when we something listed in the Asciidoc documentation, we can experiment. I'll add something to the team docs channel.
> > 
> > 
> > I've asked [here](https://redhat-internal.slack.com/archives/C02KBD8A4UF/p1709897478675109).
> 
> Hi @dfitzmau Thanks very much for adding this query to the docs channel. As discussed, I have removed the frame option.

Thanks, @sr1kar99 . Sorry to be _Sargeant Buzzkill_ but sometimes our tooling really limits things. LGTM!

radeore commented:
/lgtm

sr1kar99 commented:
> > > > > > Hi @sr1kar99 . Excellent work to you and the engineers. Discussing the limitations when limits are reached is a tough artform. I added comments inline. I like your "topbot" styling use but I'm not sure how our platform infrastructure will handle it come publication.
> > > > > > Let me know if you would like to present the styling to the wider team and get their viewpoint.
> > > > > > /remove-label peer-review-in-progress
> > > > > > /label peer-review-done
> > > > > 
> > > > > 
> > > > > Hi @dfitzmau Thank you for the review. Regarding the topbot frame, I was trying to explore the styling options, but not sure if we can use it as I thought it could lead to inconsistent styling of tables. Do we have guidelines on this?
> > > > 
> > > > 
> > > > Hi @sr1kar99 . Tbale gudiance is lacking in our [guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#formatting) doc. Usually when we something listed in the Asciidoc documentation, we can experiment. I'll add something to the team docs channel.
> > > 
> > > 
> > > I've asked [here](https://redhat-internal.slack.com/archives/C02KBD8A4UF/p1709897478675109).
> > 
> > 
> > Hi @dfitzmau Thanks very much for adding this query to the docs channel. As discussed, I have removed the frame option.
> 
> Thanks, @sr1kar99 . Sorry to be _Sargeant Buzzkill_ but sometimes our tooling really limits things. LGTM!

@dfitzmau haha! NP, Darragh! Thank you for shedding light on it. :)

sr1kar99 commented:
Hi @jakobmoellerdev @suleymanakbas91 
All the reviews for this PR are done. Before proceeding with merge review, I wanted to confirm if we can merge to 4.12+
The current documentation for Monitoring seems to be unchanged since 4.12. 
Can we apply the updates in this PR to all the branches (4.12+)..?

sr1kar99 commented:
/label merge-review-needed

adellape commented on modules/lvms-download-log-files-and-diagnostics.adoc:
```diff
@@ -8,7 +8,7 @@
 When {lvms} is unable to automatically resolve a problem, use the must-gather tool to collect the log files and diagnostic information so that you or the Red Hat Support can review the problem and determine a solution.
 
-* Run the must-gather command from the client connected to {lvms} cluster by running the following command:
+* Run the `must-gather` command from the client connected to {lvms} cluster:
 +
 [source,terminal,subs="attributes+"]
```
```suggestion
.Procedure
* Run the `must-gather` command from the client connected to {lvms} cluster:
+
[source,terminal,subs="attributes+"]
```

sr1kar99 replied:
@adellape 
Updated as per the suggestion.

adellape reviewed (commented):
Would like to hold merge on this to get a `.Procedure` added in.

adellape commented:
Please re-queue for merge-review again when ready.

sr1kar99 commented:
/label merge-review-needed

ShaunaDiaz commented:
/remove-label merge-review-needed

ShaunaDiaz commented:
/cherrypick enterprise-4.13

ShaunaDiaz commented:
/cherrypick enterprise-4.15

ShaunaDiaz commented:
/cherrypick enterprise-4.12

ShaunaDiaz commented:
@sr1kar99 1) To my knowledge, 4.12 is EOL and we are not updating those docs. 2) You can make the manual cherry picks for 4.13 and 4.14, resolve the conflicts, and either let me know, or put them back in the merge review queue and we can take care of those for you.

sr1kar99 commented:
@ShaunaDiaz I have manually cherry-picked to 4.12,  4.13, and 4.14, 4.15, and 4.16. Could you PTAL and merge the PRs?
Thanks.
